test/api_container_create_test.go

```diff
@@ -20,39 +20,99 @@ func init() {
 	SkipIfFalse(c, environment.IsLinux)
+
 	PullImage(c, busyboxImage)
 }
 
-// TestCreateOk test create api is ok with default parameters.
-func (suite *APIContainerCreateSuite) TestCreateOk(c *check.C) {
-	cname := "TestCreateOk"
+// TestBasic test create api is ok with default parameters.
+func (suite *APIContainerCreateSuite) TestBasic(c *check.C) {
+	cname := "CreateBasic"
 
 	q := url.Values{}
 	q.Add("name", cname)
 
 	obj := map[string]interface{}{
-		"Image":      busyboxImage,
-		"HostConfig": map[string]interface{}{},
+		"HostConfig": map[string]interface{}{
+			// volume related
+			"Binds": []string{
+				"/tmp:/tmp",
+			},
+
+			// runtime
+			// NOTE: please make sure the daemon has added runv
+			"Runtime": "runv",
+
+			// policy
+			"RestartPolicy": map[string]interface{}{
+				"Name": "always",
+			},
+
+			// isolation options
+			"EnableLxcfs":         true,
+			"MemoryWmarkRatio":    int64(30),
+			"MemoryExtra":         int64(50),
+			"MemoryForceEmptyCtl": 0,
+			"ScheLatSwitch":       0,
+
+			// oom setting
+			"OomScoreAdj":    100,
+			"OomKillDisable": true,
+		},
+		"Image": busyboxImage,
+		"Tty":   true,
 	}
 
 	query := request.WithQuery(q)
 	body := request.WithJSONBody(obj)
 	resp, err := request.Post("/containers/create", query, body)
 	c.Assert(err, check.IsNil)
-
 	CheckRespStatus(c, resp, 201)
 
-	// Decode response
-	got := types.ContainerCreateResp{}
-	request.DecodeBody(&got, resp.Body)
-	c.Assert(err, check.IsNil)
-	c.Assert(got.ID, check.NotNil)
+	// decode response
+	{
+		got := types.ContainerCreateResp{}
+		c.Assert(request.DecodeBody(&got, resp.Body), check.IsNil)
+		c.Assert(got.ID, check.NotNil)
+		c.Assert(got.Name, check.Equals, cname)
+		defer DelContainerForceMultyTime(c, cname)
+	}
 
-	DelContainerForceMultyTime(c, cname)
+	// check inspect result
+	{
+		resp, err := request.Get("/containers/" + cname + "/json")
+		c.Assert(err, check.IsNil)
+		CheckRespStatus(c, resp, 200)
+		got := types.ContainerJSON{}
+		c.Assert(request.DecodeBody(&got, resp.Body), check.IsNil)
+
+		// isolation related check
+		c.Assert(got.HostConfig.EnableLxcfs, check.Equals, true)
+		c.Assert(*got.HostConfig.MemoryWmarkRatio, check.Equals, int64(30))
+		c.Assert(*got.HostConfig.MemoryExtra, check.Equals, int64(50))
+		c.Assert(got.HostConfig.MemoryForceEmptyCtl, check.Equals, int64(0))
+		c.Assert(got.HostConfig.ScheLatSwitch, check.Equals, int64(0))
+
+		// runtime check
+		c.Assert(got.HostConfig.Runtime, check.Equals, "runv")
+
+		// io related check
+		c.Assert(got.Config.Tty, check.Equals, true)
+
+		// oom related check
+		c.Assert(got.HostConfig.OomScoreAdj, check.Equals, int64(100))
+		c.Assert(*got.HostConfig.OomKillDisable, check.Equals, true)
+
+		// policy
+		c.Assert(got.HostConfig.RestartPolicy.Name, check.Equals, "always")
+		c.Assert(got.HostConfig.RestartPolicy.MaximumRetryCount, check.Equals, int64(0))
+
+		// volume related
+		c.Assert(got.HostConfig.Binds, check.DeepEquals, []string{"/tmp:/tmp"})
```

Comment:
I think we had better use `c.Assert(utils.StringInSlice(got.HostConfig.Binds, "/tmp:/tmp"), check.DeepEquals, true)`.